### PR TITLE
create validate_rich_menu method

### PR DESCRIPTION
**Pull request by kenta-takeuchi**
i'd like to write a new validate rich menu api.

* Solves: #256

### CODE REVIEW

**Review by kimoto — APPROVED** [review]

File: lib/line/bot/client.rb [hunk 1]
```diff
@@ -547,6 +547,18 @@ def create_rich_menu(rich_menu)
         post(endpoint, endpoint_path, rich_menu.to_json, credentials)
       end
 
+      # Validate a rich menu
```
Comment (zenizh) [inline lib/line/bot/client.rb:550]: [nice to have]

This endpoint validates an object of the rich menu.

```suggestion
      # Validate a rich menu object
```

File: lib/line/bot/client.rb [hunk 2]
```diff
@@ -547,6 +547,18 @@ def create_rich_menu(rich_menu)
...
+      #
+      # @param rich_menu [Hash] The rich menu represented as a rich menu object
+      #
+      # @return [Net::HTTPResponse]
+      def validate_rich_menu(rich_menu)
```
Comment (zenizh) [inline lib/line/bot/client.rb:555]: [must fix]

Unlike other endpoints, this endpoint validates an object, not a rich menu. So the name should be represented more accurately.

```suggestion
      def validate_rich_menu_object(rich_menu)
```

File: spec/line/bot/rich_menu_spec.rb [hunk 3]
```diff
@@ -95,6 +95,14 @@
       .with(body: JSON.parse(RICH_MENU_CONTENT).to_json)
   end
 
+  it 'validates a rich menu' do
+    uri_template = Addressable::Template.new Line::Bot::API::DEFAULT_ENDPOINT + '/bot/richmenu/validate'
+    stub_request(:post, uri_template).to_return(body: '{}', status: 200)
+
+    client.validate_rich_menu(JSON.parse(RICH_MENU_CONTENT))
+    expect(WebMock).to have_requested(:post, Line::Bot::API::DEFAULT_ENDPOINT + '/bot/richmenu/validate')
+  end
```
Comment (zenizh) [inline spec/line/bot/rich_menu_spec.rb:104]: [must fix]

This should be changed according to the change of the method name.

```suggestion
  it 'validates a rich menu object' do
    uri_template = Addressable::Template.new Line::Bot::API::DEFAULT_ENDPOINT + '/bot/richmenu/validate'
    stub_request(:post, uri_template).to_return(body: '{}', status: 200)

    client.validate_rich_menu_object(JSON.parse(RICH_MENU_CONTENT))
    expect(WebMock).to have_requested(:post, Line::Bot::API::DEFAULT_ENDPOINT + '/bot/richmenu/validate')
  end
```

**Review by zenizh — COMMENTED** [review]
Thank you for your sending this pull request! I made a few comments, so please check them in your spare time.

File: spec/line/bot/rich_menu_spec.rb (hunk 3 above)
Comment (kenta-takeuchi) [reply spec/line/bot/rich_menu_spec.rb:104]: Thank you for your review.
I understand your comments and I will fix it later.

**Comment by kenta-takeuchi** [comment]
All comments have been addressed.
Please review them again when you have time.

**Review by zenizh — APPROVED** [review]
LGTM, thanks!

**Comment by zenizh** [comment]
@kenta-takeuchi san
FYI: If you write "Resolves #256" instead of "Solves #256" in the description, the issue is automatically closed when the pull request is merged 👍 

* EN: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
* JA: https://docs.github.com/ja/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by kenta-takeuchi** [comment]
@zenizh 
Thanks for your information.
I'll try to link issues with correct description.

**Pull request merged**